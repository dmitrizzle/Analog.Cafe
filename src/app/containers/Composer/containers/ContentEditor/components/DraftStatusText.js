// tools
import React from "react"
import styled from "styled-components"

const DraftStatusText = styled.p`
  text-align: center;
  ${props => props.theme.typography.text.auto};
  font-size: ${props => props.theme.size.font.make.normal}em;
  opacity: ${props => props.theme.opacity.half};
  margin-top: 0;
`

export default props => {
  return (
    <DraftStatusText>
      <em>{props.children}</em>
    </DraftStatusText>
  )
}
