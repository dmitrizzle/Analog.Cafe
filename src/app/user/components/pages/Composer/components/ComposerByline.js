import React from "react"

import { ROUTE_URL_USER_LANDING } from "../../../../constants/routes-session"
import Byline from "../../../../../core/components/vignettes/Byline"
import Link from "../../../../../core/components/controls/Link"

export default props => {
  return (
    <span>
      {props.user.info.role === "admin" && props.editor.status.id ? (
        <Byline>
          Submission under edit: <strong>{props.editor.status.id}</strong>{" "}
          <Link to="#unlink" onClick={props.unlinkSubmission}>
            unlink
          </Link>
          .
          {props.editor.status.type === "published" ||
          props.editor.status.type === "scheduled"
            ? [
                <br key="Byline_linebreak" />,
                <span key="BYline_note">
                  You are editing a <strong>{props.editor.status.type}</strong>{" "}
                  article.
                  {props.editor.status.type === "published" &&
                    " You will need to publish your changes to update the magazine."}
                </span>
              ]
            : null}
        </Byline>
      ) : props.user.status === "ok" ? (
        <Byline>
          <Link to={`${ROUTE_URL_USER_LANDING}/edit?return=/submit/compose`}>
            Edit
          </Link>{" "}
          your profile.
        </Byline>
      ) : (
        <Byline>
          <Link to="/sign-in">Sign in</Link> to edit your profile.
        </Byline>
      )}
    </span>
  )
}
