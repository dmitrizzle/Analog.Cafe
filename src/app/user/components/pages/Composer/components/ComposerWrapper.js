import React from "react"

import ArticleWrapper from "../../../../../core/components/pages/Article/components/ArticleWrapper"
import MetaTags from "../../../../../core/components/vignettes/MetaTags"

export default props => {
  return (
    <ArticleWrapper>
      <MetaTags
        metaTitle="Composer"
        metaDescription="A tool to upload, edit and submit your photo essays and stories."
      />
      {props.children}
    </ArticleWrapper>
  )
}
