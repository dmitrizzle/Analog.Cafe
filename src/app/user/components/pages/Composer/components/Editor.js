import { FrenchPress } from "@roast-cms/french-press-editor"
import { connect } from "react-redux"
import React from "react"

import { CARD_ERRORS } from "../../../../constants/messages-submission"
import { CARD_ERRORS as CARD_ERRORS_SESSION } from "../../../../constants/messages-session"
import { HOST_PROD } from "../../../../../constants"
import { ToggleFeature } from "../plugins"
import { setComposerSatus } from "../../../../store/actions-composer"
import { setModal } from "../../../../../core/store/actions-modal"
import CapitalA from "../../../icons/Glyphs/components/CapitalA"
import LowerA from "../../../icons/Glyphs/components/LowerA"
import Picture from "../../../../../core/components/vignettes/Picture"

class Editor extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      connectionMessageShown: false
    }
  }
  editorRef = editor => {
    this.editor = editor
  }
  handleEditorError = error => {
    if (error === "insert_image") {
      this.props.setModal(
        {
          status: "ok",
          info: CARD_ERRORS.IMAGE_SIZE(10)
        },
        { url: "errors/upload" }
      )
    }
  }
  componentWillReceiveProps = nextProps => {
    if (
      this.props.composer.focusRequested < nextProps.composer.focusRequested
    ) {
      this.editor.focus()
    }
    if (
      !this.state.connectionMessageShown &&
      nextProps.user.connection.status === "offline"
    ) {
      this.props.setModal(CARD_ERRORS_SESSION.CONNECTION_OFFLINE)
      this.setState({ connectionMessageShown: true })
    }
  }
  render = () => {
    return (
      <FrenchPress
        components={{
          Picture
        }}
        callbackStatus={this.props.setComposerSatus}
        controls={{
          MakeHeader: () => <CapitalA />,
          CancelHeader: () => <LowerA />,
          MakeQuote: () => <span>❝</span>,
          MakeLink: () => <u>link</u>,
          MakeBold: () => <strong>bold</strong>,
          MakeItalic: () => <em>italic</em>,
          UploadImage: () => <span>↫ Add Image</span>
        }}
        options={{
          domain: HOST_PROD
        }}
        callbackError={this.handleEditorError}
        editorRef={this.editorRef}
        slatePlugins={[ToggleFeature({ key: "f", node: "image" })]}
      />
    )
  }
}
const mapDispatchToProps = dispatch => {
  return {
    setModal: (info, request) => {
      dispatch(setModal(info, request))
    },
    setComposerSatus: status => {
      dispatch(setComposerSatus(status))
    }
  }
}
const mapStateToProps = state => {
  return {
    composer: state.composer,
    user: state.user
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Editor)
