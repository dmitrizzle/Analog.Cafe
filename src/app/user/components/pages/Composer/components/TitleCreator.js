import { connect } from "react-redux"
import React from "react"
import keycode from "keycode"

import {
  INPUT_SUBTITLE_LIMIT,
  INPUT_SUBTITLE_WARNING,
  INPUT_TITLE_LIMIT,
  INPUT_TITLE_WARNING
} from "../../../../constants/rules-submission"
import { resetStatus } from "../../../../../admin/store/actions-editor"
import { saveHeader } from "../../../../utils/actions-submission"
import { setComposerHeader } from "../../../../store/actions-composer"
import ComposerByline from "./ComposerByline"
import HeaderWrapper from "../../../../../core/components/vignettes/HeaderLarge/components/HeaderWrapper"
import TitleCase from "../../../forms/TitleCase"

class TitleCreator extends React.PureComponent {
  constructor(props) {
    super(props)
    this.handleTitleChange = this.handleTitleChange.bind(this)
    this.handleSubtitleChange = this.handleSubtitleChange.bind(this)
  }
  handleTitleChange = event => {
    const header = {
      title: event,
      subtitle: this.props.composer.header.subtitle
    }
    this.props.setComposerHeader(header)
    saveHeader(header)
  }
  handleSubtitleChange = event => {
    const header = {
      title: this.props.composer.header.title,
      subtitle: event
    }
    this.props.setComposerHeader(header)
    saveHeader(header)
  }
  handleKeypress = event => {
    if (keycode(event.which) === "enter") event.preventDefault()
  }
  handleUnlinkSubmission = event => {
    event.preventDefault()
    this.props.resetStatus()
  }
  render = () => {
    return (
      <HeaderWrapper>
        <TitleCase
          placeholder={this.props.pageTitle}
          onChange={this.handleTitleChange}
          value={this.props.composer.header.title}
          inputDesignation="title"
          caution={
            this.props.composer.header.title.length > INPUT_TITLE_WARNING
          }
          warning={this.props.composer.header.title.length >= INPUT_TITLE_LIMIT}
          maxLength={INPUT_TITLE_LIMIT}
          autoFocus={this.props.composer.header.title === ""}
          onKeyPress={this.handleKeypress}
        />
        <TitleCase
          placeholder={this.props.pageSubtitle}
          onChange={this.handleSubtitleChange}
          value={this.props.composer.header.subtitle}
          inputDesignation="subtitle"
          caution={
            this.props.composer.header.subtitle.length > INPUT_SUBTITLE_WARNING
          }
          warning={
            this.props.composer.header.subtitle.length >= INPUT_SUBTITLE_LIMIT
          }
          maxLength={INPUT_SUBTITLE_LIMIT}
          onKeyPress={this.handleKeypress}
        />
        <ComposerByline
          user={this.props.user}
          editor={this.props.editor}
          unlinkSubmission={this.handleUnlinkSubmission}
        />
      </HeaderWrapper>
    )
  }
}
const mapStateToProps = state => {
  return {
    composer: state.composer,
    editor: state.editor,
    user: state.user
  }
}
const mapDispatchToProps = dispatch => {
  return {
    resetStatus: () => {
      dispatch(resetStatus())
    },
    setComposerHeader: value => {
      dispatch(setComposerHeader(value))
    }
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(TitleCreator)
