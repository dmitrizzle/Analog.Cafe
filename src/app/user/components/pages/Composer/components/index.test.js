import "jest-styled-components"

import { createMockStore } from "redux-test-utils"
import { shallow } from "enzyme"
import { shallowWithStore } from "enzyme-redux"
import React from "react"

import ComposerByline from "./ComposerByline"
import ComposerWrapper from "./ComposerWrapper"
import DraftStatusText from "./DraftStatusText"
import Editor from "./Editor"
import TitleCreator from "./TitleCreator"

it("Render ComposerByline, matches snapshot", () => {
  const element = shallow(
    <ComposerByline
      user={{
        info: {
          role: "contributor"
        }
      }}
    />
  )
  expect(element).toMatchSnapshot()
})

it("Render ComposerWrapper, matches snapshot", () => {
  const element = shallow(<ComposerWrapper />)
  expect(element).toMatchSnapshot()
})

it("Render DraftStatusText, matches snapshot", () => {
  const element = shallow(<DraftStatusText />)
  expect(element).toMatchSnapshot()
})

it("Render Editor, matches snapshot", () => {
  const store = createMockStore()
  const element = shallowWithStore(<Editor />, store)
  expect(element).toMatchSnapshot()
})

it("Render TitleCreator, matches snapshot", () => {
  const store = createMockStore()
  const element = shallowWithStore(<TitleCreator />, store)
  expect(element).toMatchSnapshot()
})
