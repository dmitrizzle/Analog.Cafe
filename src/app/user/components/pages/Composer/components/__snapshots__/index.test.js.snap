// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ComposerByline, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
        Object {
            "info": Object {
              "role": "contributor",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.div>
        <Connect(NavMini)
                view="composer"
        />
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(NavMini)
          view="composer"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "view": "composer",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.div>
          <Connect(NavMini)
                    view="composer"
          />
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(NavMini)
            view="composer"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "view": "composer",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render ComposerWrapper, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          metaDescription="A tool to upload, edit and submit your photo essays and stories."
          metaTitle="Composer"
/>,
        undefined,
      ],
      "className": "fs-block",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "metaDescription": "A tool to upload, edit and submit your photo essays and stories.",
          "metaTitle": "Composer",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            metaDescription="A tool to upload, edit and submit your photo essays and stories."
            metaTitle="Composer"
/>,
          undefined,
        ],
        "className": "fs-block",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "metaDescription": "A tool to upload, edit and submit your photo essays and stories.",
            "metaTitle": "Composer",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render DraftStatusText, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <em />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "em",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <em />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "em",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render Editor, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Editor) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "composer": undefined,
      "setComposerSatus": [Function],
      "setModal": [Function],
      "user": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "composer": undefined,
        "setComposerSatus": [Function],
        "setModal": [Function],
        "user": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    },
  },
}
`;

exports[`Render TitleCreator, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(TitleCreator) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "composer": undefined,
      "editor": undefined,
      "resetStatus": [Function],
      "setComposerHeader": [Function],
      "user": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "composer": undefined,
        "editor": undefined,
        "resetStatus": [Function],
        "setComposerHeader": [Function],
        "user": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": undefined,
    },
  },
}
`;
